### PR TITLE
Use juvix-installer

### PR DESCRIPTION
**Pull request by jonaprieto**
This PR adds a loop to check if Juvix is installed, and if it's not, a notification window is shown to offer a one-click installation. It also checks if the Juvix version is outdated and offers to installer the latest.

https://github.com/anoma/vscode-juvix/assets/1428088/c36cada0-0670-4490-814b-9b4dfc3611b7

https://github.com/anoma/vscode-juvix/assets/1428088/43c8aaaf-b308-4e7b-a9e3-0d1c561ee518

### CODE REVIEW

File: src/juvixVersion.ts [hunk 1]
```diff
@@ -7,33 +7,47 @@ import * as versioning from 'compare-versions';
...
-  if (ls.status !== 0) {
-    logger.error(ERROR_JUVIX_NOT_INSTALLED);
-    return;
+  if (result === 'Install') {
+    await installJuvix();
+  } else {
+    // open the docs
+    logger.warn(
+      'Check the binary path in the configuration page or ' +
+      'visit [https://docs.juvix.org/0.3.5/howto/installing/](https://docs.juvix.org/0.3.5/howto/installing/) for instructions.'
```
Comment (vrom911) [inline src/juvixVersion.ts:33]: Can we use `${juvixVer}` here too?

**Review by vrom911 — COMMENTED** [review]
Amazing work. Tested on my mac, works smoothly 💯

File: src/installer.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  readonly shellCmd =
+    "curl --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/anoma/juvix-installer/main/juvix-installer.sh | sh && exit 0";
+
+  constructor() {
+    const options: vscode.TerminalOptions = {
+      name: 'Juvix binary installation',
+      isTransient: false,
+      hideFromUser: true,
+      env: {
+        JUVIX_INSTALLER_ASSUME_YES: '1',
```
[the remark this answers is not in the document]
Comment (jonaprieto) [reply src/installer.ts:23]: The shell script runs interactively by default and that's not needed.

File: src/juvixVersion.ts (hunk 1 above)
Comment (jonaprieto) [reply src/juvixVersion.ts:33]: Sure, let me fix this.

Comment (jonaprieto) [reply src/juvixVersion.ts]: Done

**Pull request merged**